### PR TITLE
feat(movies): add create endpoint

### PR DESCRIPTION
**Pull request by marklee22**
**What:** POST movies endpoint with movie validator logic.

**Why:** Following guide.

**Details:**
- Adding a movies POST endpoint
- Adding movies validator for title and release_year
- Adding unit tests
- Adding integration tests

### CODE REVIEW

File: lib/plugins/features/movies/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+const Controller     = require('./controller');
+const MovieValidator = require('../../../validators/movie');
```
Comment (sjl2) [inline lib/plugins/features/movies/index.js]: nit: Our general structure for validators is `validators/movies/create.js` such that it's one file per validator.

I wouldn't bother changing it now until you make a second validator though.

**Review by sjl2 — APPROVED** [review]
LGTM! Nice work.

nit about the commit message, we usually don't capitalize the first letter of the message. caps are fine when the word requires it (e.g. POST or MovieValidator).

That said, it's best to strive for as short as possible. One potential re-write for this would be `feat(movies): add create endpoint`. No need to mention the unit tests! As for rewording `POST` to create. For this, I could see it going either way, but this would be ambiguous for the update endpoint.

Also, you want to use the [imperative](https://stackoverflow.com/questions/3580013/should-i-use-past-or-present-tense-in-git-commit-messages) tense. I don't like this one as much, but someone made it a standard >.<

File: test/plugins/features/movies/controller.test.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+      return Controller.create(payload)
+        .then((movie) => {
+          expect(movie.get('title')).to.eql(payload.title);
+
+          return new Movie({ id: movie.id }).fetch();
+        })
+        .then((movie) => {
+          expect(movie.get('title')).to.eql(payload.title);
+        });
```
Comment (kienan) [inline test/plugins/features/movies/controller.test.js]: nit: shift these `.then` statements to be in line with the `return` statement.

**Review by kienan — APPROVED** [review]
LGTM once commit message addressed as described in Stewarts comment. Nice work!

File: lib/plugins/features/movies/index.js (hunk 1 above)
Comment (marklee22) [reply lib/plugins/features/movies/index.js]: Good to know. I'll make the change anyways.

What is the semantic difference between naming a file create.js versus index.js?

**Pull request merged**